analyzers/tests/SonarAnalyzer.UnitTest/Rules/Utilities/TokenTypeAnalyzerTest.Classifier.cs

```diff
@@ -980,7 +980,8 @@ public void M(Action a)
     [DataTestMethod]
     [DataRow("_ = [u:i];")]
     [DataRow("[u:i] = [u:i] + [u:i];")]
-    [DataRow("_ = i.[u:ToString]().[u:ToString]();")] // "i" must be queried. It could be a type.
+    [DataRow("_ = [u:i].[u:ToString]().[u:ToString]();")]          // Heuristic: "i" is lower case and identified as "not a type".
```

Comment:
We use a small DSL here to test the classification of tokens: `[x:token]` means: Assert that the token `token` is classified as "x". Assert that the span returned by the classification fits the location given by `[...]`. Valid values for "x" are:
<dl>
  <dt>t</dt>
  <dd>Type (class, struct, .. record, interface, enum)</dt>
  <dt>k</dt>
  <dd>Keyword (public)</dt>
  <dt>s</dt>
  <dd>String literal ("hello")</dt>
  <dt>n</dt>
  <dd>Number literal (1.0f)</dt>
  <dt>u</dt>
  <dd>Unknown (same as no-classification)</dt>
  <dt>c</dt>
  <dd>Comment (// Some comment)</dt>
</dl>

Have a look at the `ClassClassifications()` test at the beginning of the file.